### PR TITLE
feat: add adapter_type to telemetry

### PR DESCRIPTION
**Pull request by drochetti**
### Description

Adds a new property to telemetry, the `adapter_type`.

After a bit of trial and error, it seems like the correct (i.e. runtime resolved) field comes from `config.to_target_dict()["type"]`. The more obvious `config.get_metadata().adapter_type` returns the static value from the configuration.

### CODE REVIEW

File: src/fal/telemetry/telemetry.py [hunk 1]
```diff
@@ -112,6 +114,13 @@ def dbt_installed_version():
     except ImportError:
         return
 
+def get_dbt_adapter_type() -> str | None:
```
Comment (chamini2) [inline src/fal/telemetry/telemetry.py:117]: Can you instead receive the `dbt_config` in here like you are doing for adapter? Sometimes `get_dbt_config()` does not work

for example, when using

```
from fal import FalDbt

FalDbt(profiles_dir='...')
```

So we receive `dbt_config` in the `log_api` for those cases.

**Review by chamini2 — CHANGES_REQUESTED** [review]
I did not see this detail before. Otherwise good change!

I think you need to also add this in the pytest suite!

File: src/fal/telemetry/telemetry.py (hunk 1 above)
Comment (drochetti) [reply src/fal/telemetry/telemetry.py:117]: I see! That's a great suggestion, thanks =)

**Review by chamini2 — APPROVED** [review]
Looks great! Thank you for this

**Comment by drochetti** [comment]
Closed in favor of https://github.com/fal-ai/fal/pull/743

**Pull request closed without merging**